### PR TITLE
Variable width font support

### PR DESCRIPTION
**Pull request by aurelj**
Hi! Thank you for helping out with Embedded Graphics development! Please:

- [X] Check that you've added passing tests and documentation
- [x] Add a simulator example(s) where applicable
- [X] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) and appropriate crate (`embedded-graphics`, `simulator`, `tinytga`, `tinybmp`) if your changes affect the **public API**
- [X] Run `rustfmt` on the project
- [X] Run `./build.sh` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

This patch set add support for variable width fonts and add an example of such font.

Basically, a variable width font is a font in which some of the characters are not as large as the others. This allows to display some longer strings especially on low resolution displays (think small LED matrix). This can make the difference between one specific word fitting in the width of a display or not.

### CODE REVIEW

File: embedded-graphics/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,15 @@ Embedded Graphics is a `no_std` library for adding graphics features to display
...
 ## [Unreleased] - ReleaseDate
 
+### Added
+
+- Added a `6x6` hand-drawn variable width font.
+- Support for variable width fonts.
+
+### Fixed
+
+- Fonts with CHARACTER_SPACING now render properly the background color between the characters.
```
Comment (jamwaffles) [inline embedded-graphics/CHANGELOG.md]:
```suggestion
- [#267](https://github.com/jamwaffles/embedded-graphics/pull/267) Fonts with CHARACTER_SPACING now properly render the background color between characters.
```

File: embedded-graphics/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,15 @@ Embedded Graphics is a `no_std` library for adding graphics features to display
 
 ## [Unreleased] - ReleaseDate
 
+### Added
+
+- Added a `6x6` hand-drawn variable width font.
+- Support for variable width fonts.
```
Comment (jamwaffles) [inline embedded-graphics/CHANGELOG.md]:
```suggestion
- [#267](https://github.com/jamwaffles/embedded-graphics/pull/267) Added a `6x6` hand-drawn variable width font.
- [#267](https://github.com/jamwaffles/embedded-graphics/pull/267) Support for variable width fonts.
```

File: embedded-graphics/src/fonts/font6x6.rs [hunk 3]
```diff
@@ -0,0 +1,282 @@
+use crate::{fonts::Font, geometry::Size};
+
+/// 6x6 pixel variable width font.
+///
+/// [![6x6 font spritemap screenshot](https://raw.githubusercontent.com/jamwaffles/embedded-graphics/master/embedded-graphics/data/font6x6.png)](https://raw.githubusercontent.com/jamwaffles/embedded-graphics/master/embedded-graphics/data/font6x6.png)
```
Comment (jamwaffles) [inline embedded-graphics/src/fonts/font6x6.rs]: There's a table in the `src/fonts/mod.rs` module docs that needs `Font6x6` adding to it too.

File: embedded-graphics/src/fonts/text.rs [hunk 4]
```diff
@@ -115,7 +116,11 @@ where
     /// [`Size::zero()`]: ../geometry/struct.Size.html#method.zero
     fn size(&self) -> Size {
         let width = if !self.primitive.text.is_empty() {
-            (F::CHARACTER_SIZE.width + F::CHARACTER_SPACING) * self.primitive.text.len() as u32
+            self.primitive
+                .text
+                .chars()
+                .map(|c| F::char_width(c) + F::CHARACTER_SPACING)
+                .sum::<u32>()
                 - F::CHARACTER_SPACING
```
Comment (jamwaffles) [inline embedded-graphics/src/fonts/text.rs]: Thanks for finding this off-by-one :grin:

File: embedded-graphics/CHANGELOG.md (hunk 1 above)
Comment (aurelj) [reply embedded-graphics/CHANGELOG.md]: Ok, done.

File: embedded-graphics/CHANGELOG.md (hunk 2 above)
Comment (aurelj) [reply embedded-graphics/CHANGELOG.md]: Ok, done.

File: embedded-graphics/src/fonts/font6x6.rs (hunk 3 above)
Comment (aurelj) [reply embedded-graphics/src/fonts/font6x6.rs]: Oh, I missed this one indeed. Fixed.

File: embedded-graphics/src/fonts/text.rs (hunk 4 above)
Comment (aurelj) [reply embedded-graphics/src/fonts/text.rs]: :smile:

**Comment by aurelj** [comment]
> Looks good aside from a couple of nits. Can you also add an example to the simulator please? This could either append to the `text-fonts.rs` example, or add a new `text-variable-width.rs` example.

Sure, I just added an example in `text-fonts.rs`.
